src/hooks/useFetchEASData.ts

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from "react";
+
+import axios from "axios";
+
+const useFetchEASData = (query, variables) => {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [data, setData] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setLoading(true);
```

Comment:
Must check if loading is tru before entering fetchData to avoid multiple loadings

Reply:
Sure, thank you for the feedback. Please check the changes to see if it's correct!